tests/wallet/nft_wallet/test_nft_mint_from_did.py

```diff
@@ -0,0 +1,494 @@
+
+    did_wallet_maker: DIDWallet = await DIDWallet.create_new_did_wallet(
+        wallet_node_maker.wallet_state_manager, wallet_maker, uint64(1)
+    )
+    spend_bundle_list = await wallet_node_maker.wallet_state_manager.tx_store.get_unconfirmed_for_wallet(
+        did_wallet_maker.id()
+    )
+
+    spend_bundle = spend_bundle_list[0].spend_bundle
+    await time_out_assert_not_none(5, full_node_api.full_node.mempool_manager.get_spendbundle, spend_bundle.name())
+
+    await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph_token))
+
+    await time_out_assert(15, wallet_maker.get_pending_change_balance, 0)
+    await time_out_assert(10, wallet_maker.get_unconfirmed_balance, funds - 1)
+    await time_out_assert(10, wallet_maker.get_confirmed_balance, funds - 1)
+
+    hex_did_id = did_wallet_maker.get_my_DID()
+    hmr_did_id = encode_puzzle_hash(bytes32.from_hexstr(hex_did_id), AddressType.DID.hrp(config))
+
+    nft_wallet_maker = await api_maker.create_new_wallet(
+        dict(wallet_type="nft_wallet", name="NFT WALLET 1", did_id=hmr_did_id)
+    )
+    assert isinstance(nft_wallet_maker, dict)
+    assert nft_wallet_maker.get("success")
+
+    nft_wallet_taker = await api_taker.create_new_wallet(dict(wallet_type="nft_wallet", name="NFT WALLET 2"))
+
+    await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph_token))
+
+    sample = {
+        "hash": bytes32(token_bytes(32)).hex(),
+        "uris": ["https://data.com/1234"],
+        "meta_hash": bytes32(token_bytes(32)).hex(),
+        "meta_uris": ["https://meatadata.com/1234"],
+        "license_hash": bytes32(token_bytes(32)).hex(),
+        "license_uris": ["https://license.com/1234"],
+        "series_numer": 1,
+        "series_total": 1,
+    }
+
+    try:
+        n = 10
+        metadata_list = [sample for x in range(n)]
+        target_list = [encode_puzzle_hash((ph_taker), "xch") for x in range(n)]
+        royalty_address = encode_puzzle_hash(bytes32(token_bytes(32)), "xch")
+        royalty_percentage = 300
+        fee = 100
+        required_amount = n + (fee * n)
+        xch_coins = await client.select_coins(amount=required_amount, wallet_id=wallet_maker.id())
+        funding_coin = xch_coins[0]
+        assert funding_coin.amount >= required_amount
+        funding_coin_dict = xch_coins[0].to_json_dict()
+        chunk = 5
+        next_coin = funding_coin
+        did_coin = (await client.select_coins(amount=1, wallet_id=2))[0]
+        did_lineage_parent = None
+        spends = []
+
+        for i in range(0, n, chunk):
+            resp: Dict[str, Any] = await client.nft_mint_from_did(
+                wallet_id=nft_wallet_maker["wallet_id"],
+                metadata_list=metadata_list[i : i + chunk],
+                target_list=target_list[i : i + chunk],
+                royalty_percentage=royalty_percentage,
+                royalty_address=royalty_address,
+                starting_num=i + 1,
+                max_num=n,
```

Comment:
I don't get why `max_num=n`. Shouldn't it be `max_num=chunk`? Because `nft_mint_from_did` gets a sliced metadata_list (of length `chunk`) a few lines above, not the complete one. Also, why is `starting_num` not 1? All confusing to me.
This test only checks for correct amount of minted NFTs. Would be nice if it would also confirm they have in fact the correct metadata (of course don't use the same metadata for each as it is now).

Reply:
max_num is always fixed as the total number of NFTs to be minted, equivalent to edition_total.  Admittedly the indexing is a bit confusing,  but basically we give the function a starting num and a max num, and within the minting function we then calculate the other values needed so we can create the ephemeral puzzles with unique m-of-n values. 
Have updated the test (see comment below)

Reply:
Are you aware of the recent definition clarifications of `series` and `edition` as outlined in the [CHIP 005](https://github.com/Chia-Network/chips/blob/nft1/CHIPs/chip-0005.md)? I believe the overwhelming majority of NFTs (including Chia Friends) intend(ed) to have `edition_number` and `edition_total` set to **1** for all NFTs. I would be surprised if "max_num is always fixed as the total number of NFTs to be minted, equivalent to edition_total" as this means single mints as before.

Reply:
Actually I believe this minting method should not care at all what the user supplies as `edition_number` and `edition_total`. It should just pass it on, which it does (it's contained of the metadata list given to this method), so that's fine. The above comment is just for clarification.